### PR TITLE
Card preview

### PR DESCRIPTION
**Pull request by JanikOskar**
@laniszew @lmazurowski

### CODE REVIEW

File: src/components/CardPreview/CardPreview.tsx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+`;
+
+const callPersonaMutation = (
+  action: string,
+  personaId: string,
+  selectedPersonaId: string
+): Promise<ExecutionResult<RecommendPersonaResponse | SavePersonaResponse>> | null => {
+  const savePersona = partial(client.mutate, { mutation: SAVE_PERSONA });
+  const recommendPersona = partial(client.mutate, { mutation: RECOMMEND_PERSONA });
+  switch (action) {
```
Comment (lmazurowski) [inline src/components/CardPreview/CardPreview.tsx]: one line space, please :)

File: src/components/CardPreview/CardPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+  }
+  const handleSetDefault = async (uuid: string): Promise<void> => {
+    try {
+      const { data } = await setDefaultPersona({ variables: { uuid: uuid } });
+      setDefaultPersonaUuid(data?.setDefaultPersona.uuid);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+  return (
```
Comment (lmazurowski) [inline src/components/CardPreview/CardPreview.tsx]: one line space, please :)

File: src/components/CardPreview/CardPreview.tsx [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    <div>
```
Comment (lmazurowski) [inline src/components/CardPreview/CardPreview.tsx]: <div> is necessary, or we can go with react fragment here?

File: src/components/CardPreview/CardPreview.tsx [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+      <Overlay>
+        <Spinner />
+      </Overlay>
+    );
+  }
+  // OR !data is used cause typescript doesn't know that data can no longer be undefined in return method
+  if (isEmpty(data?.userPersonas) || !data) {
+    return <div>No personas...</div>;
+  }
+  const handleSetDefault = async (uuid: string): Promise<void> => {
```
Comment (lmazurowski) [inline src/components/CardPreview/CardPreview.tsx]: one line space, please

File: src/components/CardPreview/CardPreview.tsx [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+      const defaultPersonaIndex = data.userPersonas.findIndex(persona => persona.uuid === defaultPersonaUuid);
+      const goToIndex = defaultPersonaIndex === -1 ? 0 : defaultPersonaIndex;
+      setCurrentSlide(goToIndex);
+      carousel.current?.goTo(goToIndex);
+    }
+  }, [data, defaultPersonaUuid]);
+
+  const handleShare = (): void => {
+    const pathnameArray = getPathnameArray(pathname);
+    if (!data) {
```
Comment (lmazurowski) [inline src/components/CardPreview/CardPreview.tsx]: give it some space here in handleShare for better readibility

**Review by laniszews — APPROVED** [review]

**Review by laniszew — APPROVED** [review]

**Pull request merged**